File: publicq-main/client/src/models/smtp-options.ts
/**
 * SMTP configuration options for email delivery
 */
export interface SmtpOptions {
  /**
   * SMTP host address
   */
  smtpHost: string;

  /**
   * SMTP port number
   */
  smtpPort: number;

  /**
   * Optional: username for SMTP authentication
   */
  userName?: string;

  /**
   * Optional: password for SMTP authentication
   */
  password?: string;

  /**
   * Use STARTTLS if the server supports it
   */
  useStartTls: boolean;

  /**
   * Use SSL to connect to the SMTP server
   */
  useSsl: boolean;
}

File: publicq-main/src/PublicQ.Infrastructure/Options/SmtpOptions.cs
namespace PublicQ.Infrastructure.Options;

/// <summary>
/// SMTP configuration
/// </summary>
public class SmtpOptions
{
    /// <summary>
    /// SMTP host address.
    /// </summary>
    public string SmtpHost { get; set; } = string.Empty;
    
    /// <summary>
    /// Smtp port number.
    /// </summary>
    public int SmtpPort { get; set; }
    
    /// <summary>
    /// Optional: username for SMTP authentication.
    /// </summary>
    public string? UserName { get; set; }
    
    /// <summary>
    /// Optional: password for SMTP authentication.
    /// </summary>
    public string? Password { get; set; }
    
    /// <summary>
    /// Use STARTTLS if the server supports it.
    /// </summary>
    public bool UseStartTls { get; set; }
    
    /// <summary>
    /// Use SSL to connect to the SMTP server.
    /// </summary>
    public bool UseSsl { get; set; }
}